ios/Classes/codec/AblyFlutterReader.m

```diff
@@ -113,9 +112,28 @@ -(ARTClientOptions *)readClientOptions {
     READ_VALUE(o, defaultTokenParams);
     [self readValue]; // TODO READ_VALUE(o, channelRetryTimeout); // NSTimeInterval
     [self readValue]; // TODO READ_VALUE(o, transportParams); // property not found
-    [self readValue]; // TODO READ_VALUE(o, asyncHttpThreadpoolSize); // property not found
-    READ_VALUE(o, pushFullWait);
+    //    [self readValue]; // TODO READ_VALUE(o, asyncHttpThreadpoolSize); // property not found
+    //    READ_VALUE(o, pushFullWait);
+    
+    return o;
+}
+
+
+-(ARTTokenDetails *)readTokenDeatils {
+    __block NSString *token = nil;
+    __block NSDate *expires = nil;
+    __block NSDate *issued = nil;
+    __block NSString *capability = nil;
+    __block NSString *clientId = nil;
+    
+    ON_VALUE(^(const id value) { token = value; })
+    ON_VALUE(^(const id value) { expires = value; })
+    ON_VALUE(^(const id value) { issued = value; })
+    ON_VALUE(^(const id value) { capability = value; })
+    ON_VALUE(^(const id value) { clientId = value; })
```

Comment:
@mattheworiordan my thoughts on this:

If other developers are working on `ably-flutter` package, I suppose they will be handling encoding and decoding on all 3 ends (iOS, dart and Android). Even if not, ordering can be decided and must be followed on all 3 implementations of codec.

Reply:
@mattheworiordan you are, of course, right. However, Rohit and I have had quite a long conversation about this earlier today. Akin to [my earlier comment](https://github.com/ably/ably-flutter/pull/5#discussion_r419369977), this counts as an implementation detail that's contained within this codebase and doesn't affect the public API.

For that reason I would rather tackle it when we need to, if it becomes an issue. Essentially the desire would be to have some kind of a map rather than sequential serialisation of values, with the focus being on reducing the 'brittle' from a maintenance perspective.

There is, however, also the argument that this method of serialisation will be quicker at runtime, without the need to look up in wrappers that provide us maps. That is an unproven point, as we're not measuring performance at the moment, but basic sequential serialisation of values 'in order' is the default out-of-the-box approach promoted by the plugin examples presented by the Flutter developers themselves.

We can optimise for performance or we can optimise for maintainability, but either way I don't see a pressing reason why either of those concerns is banging down the door right now for attention.

Reply:
> For that reason I would rather tackle it when we need to, if it becomes an issue.

Ok, well I would prefer to approach problems to avoid issues we can see down the road.

> Essentially the desire would be to have some kind of a map rather than sequential serialisation of values, with the focus being on reducing the 'brittle' from a maintenance perspective.

Sounds great, any reason not to tackle this now to avoid the problem becoming bigger in every PR (every PR has 3 copies of this problem)

> There is, however, also the argument that this method of serialisation will be quicker at runtime, without the need to look up in wrappers that provide us maps. T

We are not optimising for performance.  We have never done this in SDKs, and we wouldn't be building a wrapper if this was the case.

> We can optimise for performance or we can optimise for maintainability,

We ALWAYS opt for maintainability, ease of debugging, and then when we need to opt for performance.  I can't think of any reason why we would opt for performance in a wrapper SDK, when users will be blocked on IO, not CPU cycles passing buffers.

> I don't see a pressing reason why either of those concerns is banging down the door right now for attention.

Because every PR makes this a bigger problem to fix later.  And a missing line or incorrect ordering of buffers, may go undetected, and result in a tremendously difficult bug to fix in 6 months time when this is used in production. Debugging this across Dart -> iOS/Java, is going to be very hard, and we can avoid this problem now with a conscious decision to address it.

Reply:
Fair enough. @tiholic is going to work on this as you request.

Reply:
Thanks 👍

Reply:
P.S. I don't monitor all the PRs, so please do tell me when a solution exists.  Interested to see what you come up with for this.